lib/promoted/ruby/client.rb

```diff
@@ -262,6 +266,22 @@ def send_log_request log_request_params, headers={}
 
         private
 
+        def do_warmup
+          if !@delivery_endpoint
+            # Warmup only supported when delivery is enabled.
+            return
+          end
+
+          warmup_url = @delivery_endpoint.reverse.sub("/deliver".reverse, "/healthz".reverse).reverse
+          @logger.info("Warming up at #{warmup_url}") if @logger
+          1.upto(20) do
```

Comment:
Is it easy to refactor out the constant for max_threads and reuse with a multiple?

Reply:
Looking at the net-http-persistent code, the default pool size is a fraction of the system's file handles, which I don't want to override or poke around to find. 10 seems to work, 20 seems safe. Overall this takes around half a second outside-region.